docs/concepts-advanced/when-clauses.md

```diff
@@ -0,0 +1,125 @@
+
+## Functions
+
+The following functions are available for handling variables safely:
+
+| Function | Description | Example |
+|----------|-------------|---------|
+| `withDefault(variable, default)` | Returns the variable value if it exists, otherwise returns the default value | `withDefault("NONEXISTENT_VAR", false) == true` |
+| `exists(variable)` | Returns true if the variable exists, false otherwise | `exists("LAGOON_GIT_BRANCH")` |
+
+### Using Functions Safely
+
+The `withDefault()` and `exists()` functions are particularly useful for preventing errors when referencing variables that may not exist in all environments:
+
+- Use `exists()` to check if a variable is defined before using it
+- Use `withDefault()` to provide fallback values for undefined variables
+- These functions prevent panics that can occur when referencing undefined variables
+
+## Usage Examples
+
+### Basic Comparisons
+
+```yaml title=".lagoon.yml"
+tasks:
+  post-rollout:
+    - run:
+        name: Run on main branch only
+        command: drush cim
+        service: cli
+        when: LAGOON_GIT_BRANCH == "main"
+    - run:
+        name: Skip on production
+        command: drush sql-sync @prod @self
+        service: cli
+        when: LAGOON_ENVIRONMENT_TYPE != "production"
+```
+
+### Regex Pattern Matching
+
+```yaml title=".lagoon.yml"
+tasks:
+  post-rollout:
+    - run:
+        name: Run on feature branches
+        command: ./scripts/feature-setup.sh
+        service: cli
+        when: LAGOON_GIT_BRANCH =~ "^feature/.*"
+    - run:
+        name: Skip hotfix branches
+        command: ./scripts/standard-deploy.sh
+        service: cli
+        when: LAGOON_GIT_BRANCH !~ "^hotfix/.*"
+```
+
+### Variable Existence and Defaults
+
+```yaml title=".lagoon.yml"
+tasks:
+  post-rollout:
+    - run:
+        name: Run only if custom variable exists
+        command: ./scripts/custom-setup.sh
+        service: cli
+        when: exists("CUSTOM_DEPLOY_FLAG")
+    - run:
+        name: Use default value for undefined variable
+        command: echo "Debug mode enabled"
+        service: cli
+        when: withDefault("DEBUG_MODE", false) == true
+```
+
+### Complex Conditions
+
+```yaml title=".lagoon.yml"
+tasks:
+  post-rollout:
+    - run:
+        name: Complex condition with default
+        command: ./scripts/conditional-task.sh
+        service: cli
+        when: withDefault("FEATURE_ENABLED", "false") == "true" && LAGOON_GIT_BRANCH == "main"
+    - run:
+        name: Multiple conditions
+        command: ./scripts/staging-sync.sh
+        service: cli
+        when: LAGOON_ENVIRONMENT_TYPE == "development" && exists("SYNC_FROM_PROD")
+```
+
+## Available Variables
+
+You can reference any of the following in `when` clauses:
+
+- Default Lagoon environment variables (such as `LAGOON_GIT_BRANCH`, `LAGOON_ENVIRONMENT_TYPE`, etc.)
+- Any `BUILD` or `GLOBAL` scoped variables defined in the Lagoon API
```

Comment:
This should also evaluate `RUNTIME` too then no? Reading the code in the build-deploy-tool PR, it passes in the lagoon api envvars, so this would include `RUNTIME` in it too?